### PR TITLE
POC for deleteOne and deleteMany for tables

### PR DESCRIPTION
**Pull request by amorton**
**What this PR does**:

Adds a new DeleteTableOperation and re-uses the filter resolving, and resolver framework for tables.
    
Note: we cannot know the number of records deleted, and we cannot limit the number - unless we validate that the entire PK was supplied. Will check those things when we get to Beta,

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [*] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [*] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/DeleteTableOperation.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    // Delete in CQL allows you to delete columns, we do not support that, next to add is the WHERE
+    // clause
+    // HACK AARON - do not know how to get on the correct interface here
+    Delete delete = (Delete) deleteSelection;
+    List<Object> positionalValues = new ArrayList<>();
+    delete = whereCQLClause.apply(delete, positionalValues);
+
+    var statement = delete.build(positionalValues.toArray());
+
+    LOGGER.warn("DELETE CQL: {}", delete.asCql());
```
Comment (Yuqi-Du) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/DeleteTableOperation.java:54]: Maybe change to Info or just delete the the log if these are just added for debug purpuse.

**Review by Yuqi-Du — APPROVED** [review]
LGTM, left one suggestion

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/DeleteTableOperation.java (hunk 1 above)
Comment (amorton) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/DeleteTableOperation.java:54]: def will do once we get out of alpha, there are they  to make it easy to see the CQL we are generating

**Comment by jeffreyscarpenter** [comment]
@amorton what is the status of this?

**Review by vkarpov15 — COMMENTED** [review]
LGTM

**Pull request merged**